### DISCUSSION TITLE
Magit v4.0.0 released

### DISCUSSION THREAD

**Discussion by tarsius** [magit/magit#5194] [Announcements]
* [Announcement](https://emacsair.me/2024/08/09/magit-4.0)
* [Release Notes](https://github.com/magit/magit/blob/main/docs/RelNotes/4.0.0.org)

<hr /><em>This discussion was created from the release <a href='https://github.com/magit/magit/releases/tag/v4.0.0'>v4.0.0</a>.</em>

**Comment by aylons** [comment]
Hi, the link to the announcement is broken! I believe the date is incorrect. Same thing for the link in the release notes.

FOr anyone reading this before he fixes: [Announcement](https://emacsair.me/2024/08/09/magit-4.0)

**Reply by tarsius** [reply]
Thanks, fixed.

**Comment by waldencui** [comment]
Congratulations for the new release! 

I see the following when opening \*Help\* for Magit in \*Packages\*:
> describe-package-1: Wrong type argument: char-or-string-p, ("Jonas Bernoulli" . "emacs.magit@jonas.bernoulli.dev")

**Reply by tarsius** [reply]
Thanks!

Some older Emacs releases had a bug here. Not sure when it was fixed, but I just tried 29.1 and there it works.

**Reply by waldencui** [reply]
I'm using Emacs 29.4 BTW.

**Reply by tarsius** [reply]
The reason I couldn't reproduce it is that I have added MELPA to `package-archives` and that produces an `archive-contents`, which is backward compatible. Without that entry, `nongnu` is used, and its `package-archives` file unfortunately is not backward compatible. The `:maintainer` property is the problem.

<details>

```elisp
;;; Melpa
(magit . [(20240811 1419)
          ((emacs (26 1))
           (compat (30 0 0 0))
           (dash (20240510))
           (git-commit (20240808))
           (magit-section (20240808))
           (seq (2 24))
           (transient (20240805))
           (with-editor (20240806)))
          "A Git porcelain inside Emacs." tar
          ((:commit . "a2739d7db1fdf19b95f36f6ddd15b0c1f523bd26")
           (:authors ("Marius Vollmer" . "marius.vollmer@gmail.com")
                     ("Jonas Bernoulli" . "emacs.magit@jonas.bernoulli.dev"))
           (:maintainer "Jonas Bernoulli" . "emacs.magit@jonas.bernoulli.dev")
           (:keywords "git" "tools" "vc")
           (:url . "https://github.com/magit/magit"))])

;;; NonGNU ELPA
(magit .
       [(4 0 0)
        ((emacs
          (26 1))
         (compat
          (30 0 0 0))
         (dash
          (2 19 1))
         (git-commit
          (4 0 0))
         (magit-section
          (4 0 0))
         (seq
          (2 24))
         (transient
          (0 7 4))
         (with-editor
           (3 4 1)))
        "A Git porcelain inside Emacs" tar
        ((:url . "https://github.com/magit/magit")
         (:keywords "git" "tools" "vc")
         (:maintainer
          ("Jonas Bernoulli" . "emacs.magit@jonas.bernoulli.dev")
          ("Kyle Meyer" . "kyle@kyleam.com"))
         (:authors
          ("Marius Vollmer" . "marius.vollmer@gmail.com")
          ("Jonas Bernoulli" . "emacs.magit@jonas.bernoulli.dev"))
         (:commit . "020aca7c9c4154dbc4a72acbd56165ecccea1bf1"))])
```

</details>


I'll talk to the `package` and `elpa-admin` maintainers about this (again), but that could take a while.
